pyspark/bigdl/models/rnn/rnnexample.py

```diff
@@ -0,0 +1,171 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Still in experimental stage!
+
+import itertools
+import re
+from optparse import OptionParser
+
+from bigdl.dataset import sentence
+from bigdl.nn.layer import *
+from bigdl.nn.criterion import *
+from bigdl.optim.optimizer import *
+from bigdl.util.common import *
+from bigdl.util.common import Sample
+
+def prepare_data(sc, folder, vocabsize):
+    train_file = folder + "train.txt"
+    val_file = folder + "val.txt"
+
+    train_sentences_rdd = sc.textFile(train_file) \
+        .map(lambda line: sentence.sentences_split(line))
+    pad_sent = train_sentences_rdd.flatMap(lambda x: x). \
+        map(lambda sent: sentence.sentences_bipadding(sent))
+    tokens = pad_sent.map(lambda pad: sentence.sentence_tokenizer(pad))
+    max_len = tokens.map(lambda x: len(x)).max()
+    print("max length %s" % max_len)
+
+    words = tokens.flatMap(lambda x: x)
+    print("%s words and %s sentences processed in train data" % (words.count(), tokens.count()))
+
+    val_sentences_rdd = sc.textFile(val_file) \
```

Comment:
should we extract this as a method? seems like there are some duplicate code between train&val dataset